pkg/network/networker.go

```diff
@@ -156,28 +157,54 @@ func (n *networker) attachYgg(id string, netNs ns.NetNS) (net.IPNet, error) {
 		return net.IPNet{}, fmt.Errorf("failed to generate ygg subnet IP: %w", err)
 	}
 
-	ips := []*net.IPNet{
-		&ip,
-	}
-
 	gw, err := n.ygg.Gateway()
 	if err != nil {
 		return net.IPNet{}, fmt.Errorf("failed to get ygg gateway IP: %w", err)
 	}
 
-	routes := []*netlink.Route{
-		{
-			Dst: &net.IPNet{
-				IP:   net.ParseIP("200::"),
-				Mask: net.CIDRMask(7, 128),
-			},
-			Gw: gw.IP,
-			// LinkIndex:... this is set by macvlan.Install
+	route := netlink.Route{
+		Dst: &net.IPNet{
+			IP:   net.ParseIP("200::"),
+			Mask: net.CIDRMask(7, 128),
 		},
+		Gw: gw.IP,
 	}
 
-	if err := n.createMacVlan(ZDBYggIface, types.YggBridge, hw, ips, routes, netNs); err != nil {
-		return net.IPNet{}, errors.Wrap(err, "failed to setup zdb ygg interface")
+	name := filepath.Base(netNs.Path())
+	peerPrefix := name
+	if len(name) > 4 {
+		peerPrefix = name[0:4]
+	}
+
+	if !ifaceutil.Exists(ZDBYggIface, netNs) {
+		yggLink, err := ifaceutil.MakeVethPair(ZDBYggIface, types.YggBridge, 1500, peerPrefix)
+		if err != nil {
+			return net.IPNet{}, fmt.Errorf("failed to create ygg link: %w", err)
+		}
+
+		err = netlink.LinkSetNsFd(yggLink, int(netNs.Fd()))
+		if err != nil {
+			return net.IPNet{}, fmt.Errorf("failed to move ygg link: %s to namespace:%s : %w", ZDBYggIface, netNs.Path(), err)
+		}
+
+		if err := netNs.Do(func(_ ns.NetNS) error {
+			err = setLinkAddr(ZDBYggIface, &ip)
+			if err != nil {
+				return err
+			}
+
+			if err := ifaceutil.SetLoUp(); err != nil {
+				return fmt.Errorf("failed to set lo up for namespace '%s': %w", name, err)
+			}
+
+			if err := options.SetIPv6Forwarding(true); err != nil {
+				return fmt.Errorf("failed to enable ipv6 forwarding in namespace %q: %w", name, err)
+			}
+
+			return netlink.RouteAdd(&route)
+		}); err != nil {
+			return net.IPNet{}, fmt.Errorf("failed to setup link addresses %s: %w", ZDBYggIface, err)
+		}
```

Comment:
Looks like this code block does the same thing as https://github.com/threefoldtech/zos/pull/2414/files#diff-4e15b6f2760eb52c8292eadf92ce4a6ba3afcf08d34b528d56081b9bc1623554R252-R285

extract it to a func?